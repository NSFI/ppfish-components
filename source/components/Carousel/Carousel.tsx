import * as React from 'react';
import { findDOMNode } from 'react-dom';
import debounce from 'lodash/debounce';
import classNames from 'classnames';

// matchMedia polyfill for
// https://github.com/WickyNilliams/enquire.js/issues/82
if (typeof window !== 'undefined') {
  const matchMediaPolyfill = (mediaQuery: string) => {
    return {
      media: mediaQuery,
      matches: false,
      addListener() {},
      removeListener() {},
    } as any as MediaQueryList;
  };
  window.matchMedia = window.matchMedia || matchMediaPolyfill;
}
// Use require over import (will be lifted up)
// make sure matchMedia polyfill run before require('react-slick')
// Fix https://github.com/ant-design/ant-design/issues/6560
// Fix https://github.com/ant-design/ant-design/issues/3308
const SlickCarousel = require('react-slick').default;

export type CarouselEffect = 'scrollx' | 'fade';
export type DotsPosition = 'top' | 'right' | 'bottom' | 'left';

// Carousel
export interface CarouselProps {
  effect?: CarouselEffect;
  dots?: boolean;
  autoplay?: boolean;
  easing?: string;
  beforeChange?: (from: number, to: number) => void;
  afterChange?: (current: number) => void;
  style?: React.CSSProperties;
  prefixCls?: string;
  accessibility?: boolean;
  dotsTimer?: boolean;
  dotsPosition?: DotsPosition;
  nextArrow?: HTMLElement | any;
  prevArrow?: HTMLElement | any;
  pauseOnHover?: boolean;
  className?: string;
  adaptiveHeight?: boolean;
  arrows?: boolean;
  autoplaySpeed?: number;
  centerMode?: boolean;
  centerPadding?: string | any;
  cssEase?: string | any;
  dotsClass?: string;
  draggable?: boolean;
  fade?: boolean;
  focusOnSelect?: boolean;
  infinite?: boolean;
  initialSlide?: number;
  lazyLoad?: boolean;
  rtl?: boolean;
  slide?: string;
  slidesToShow?: number;
  slidesToScroll?: number;
  speed?: number;
  swipe?: boolean;
  swipeToSlide?: boolean;
  touchMove?: boolean;
  touchThreshold?: number;
  variableWidth?: boolean;
  useCSS?: boolean;
  slickGoTo?: number;
  children?: React.ReactNode;
}

function CustomArrow(props) {
  const { className, style, onClick } = props;
  return <div className={className} style={{ ...style, display: 'flex' }} onClick={onClick} />;
}

export interface CarouselRef {
  goTo: (slide: number, dontAnimate?: boolean) => void;
  next: () => void;
  prev: () => void;
  autoPlay: boolean;
  innerSlider: any;
}

const InternalCarousel: React.ForwardRefRenderFunction<CarouselRef, CarouselProps> = (
  props,
  ref,
) => {
  const slickRef = React.useRef<any>();
  React.useImperativeHandle(
    ref,
    () => ({
      goTo,
      autoPlay: slickRef.current.innerSlider.autoPlay,
      innerSlider: slickRef.current.innerSlider,
      prev,
      next,
    }),
    [slickRef.current],
  );

  const goTo = (slide: number, dontAnimate = false) => {
    slickRef.current.slickGoTo(slide, dontAnimate);
  };

  const prev = () => {
    slickRef.current.slickPrev();
  };

  const next = () => {
    slickRef.current.slickNext();
  };

  React.useEffect(() => {
    const { autoplay, autoplaySpeed, dotsPosition, dotsTimer } = props;
    const innerSlider = slickRef.current && slickRef.current.innerSlider;

    let fn = () => {
      const { autoplay } = props;
      if (autoplay && innerSlider && slickRef.current.innerSlider.autoPlay) {
        slickRef.current.innerSlider.autoPlay();
      }
    };

    const slickDOM = findDOMNode(slickRef.current) as HTMLElement;
    if (autoplay && dotsTimer) {
      let aniName =
          dotsPosition == 'left' || dotsPosition == 'right' ? 'dotsAniVertical' : 'dotsAni',
        timerEl = slickDOM.querySelector('.timer') as HTMLElement;
      !!timerEl &&
        timerEl.style.setProperty('--dots-ani', `${aniName} ${autoplaySpeed / 1000}s infinite`);

      slickDOM.addEventListener('mouseover', () => {
        let timerEl = slickDOM.querySelector('.timer') as HTMLElement;
        !!timerEl && timerEl.style.setProperty('--dots-ani', 'none');
      });

      slickDOM.addEventListener('mouseout', () => {
        let timerEl = slickDOM.querySelector('.timer') as HTMLElement;
        !!timerEl &&
          timerEl.style.setProperty('--dots-ani', `${aniName} ${autoplaySpeed / 1000}s infinite`);
      });
    }
    if (autoplay) {
      const onWindowResized = debounce(fn, 500, {
        leading: false,
      });
      window.addEventListener('resize', onWindowResized);

      return () => {
        window.removeEventListener('resize', onWindowResized);
        (onWindowResized as any).cancel();
      };
    }
  }, [slickRef.current, props.autoplay]);

  let {
    prefixCls,
    className,
    style,
    dotsPosition,
    dotsTimer,
    nextArrow,
    prevArrow,
    centerMode,
    centerPadding,
    slidesToShow,
    ...restProps
  } = props;
  let cls = classNames(prefixCls, `${prefixCls}-${dotsPosition}`, className);
  let dotsCls = classNames('slick-dots', {
    'slick-dots-vertical': dotsPosition == 'left' || dotsPosition == 'right',
    timer: restProps.autoplay && dotsTimer,
  });

  if (restProps.effect === 'fade') {
    restProps.fade = true;
  }

  return (
    <div className={cls} style={style}>
      <SlickCarousel
        {...restProps}
        ref={slickRef}
        nextArrow={nextArrow ? nextArrow : <CustomArrow className="slick-next" onClick={next} />}
        prevArrow={prevArrow ? prevArrow : <CustomArrow className="slick-prev" onClick={prev} />}
        centerMode={centerMode}
        centerPadding={centerPadding}
        slidesToShow={slidesToShow}
        dotsClass={dotsCls}
      />
    </div>
  );
};

const Carousel = React.forwardRef<CarouselRef, CarouselProps>(InternalCarousel);

Carousel.displayName = 'Carousel';

Carousel.defaultProps = {
  dots: true,
  arrows: false,
  prefixCls: 'fishd-carousel',
  draggable: false,
  dotsTimer: false,
  autoplaySpeed: 3000,
  dotsPosition: 'bottom',
  centerMode: false,
  centerPadding: '50px',
  slidesToShow: 1,
};

export default Carousel;
