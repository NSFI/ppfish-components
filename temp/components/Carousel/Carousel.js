var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import * as React from 'react';
import { findDOMNode } from 'react-dom';
import debounce from 'lodash/debounce';
import classNames from 'classnames';
// matchMedia polyfill for
// https://github.com/WickyNilliams/enquire.js/issues/82
if (typeof window !== 'undefined') {
    var matchMediaPolyfill = function (mediaQuery) {
        return {
            media: mediaQuery,
            matches: false,
            addListener: function () { },
            removeListener: function () { }
        };
    };
    window.matchMedia = window.matchMedia || matchMediaPolyfill;
}
// Use require over import (will be lifted up)
// make sure matchMedia polyfill run before require('react-slick')
// Fix https://github.com/ant-design/ant-design/issues/6560
// Fix https://github.com/ant-design/ant-design/issues/3308
var SlickCarousel = require('react-slick').default;
function CustomArrow(props) {
    var className = props.className, style = props.style, onClick = props.onClick;
    return React.createElement("div", { className: className, style: __assign(__assign({}, style), { display: 'flex' }), onClick: onClick });
}
var Carousel = /** @class */ (function (_super) {
    __extends(Carousel, _super);
    function Carousel(props) {
        var _this = _super.call(this, props) || this;
        _this.onWindowResized = function () {
            // Fix https://github.com/ant-design/ant-design/issues/2550
            var autoplay = _this.props.autoplay;
            if (autoplay && _this.slick && _this.slick.innerSlider && _this.slick.innerSlider.autoPlay) {
                _this.slick.innerSlider.autoPlay();
            }
        };
        _this.saveSlick = function (node) {
            _this.slick = node;
        };
        _this.onWindowResized = debounce(_this.onWindowResized, 500, {
            leading: false
        });
        return _this;
    }
    Carousel.prototype.componentDidMount = function () {
        var _this = this;
        var _a = this.props, autoplay = _a.autoplay, autoplaySpeed = _a.autoplaySpeed, dotsPosition = _a.dotsPosition, dotsTimer = _a.dotsTimer;
        if (autoplay) {
            window.addEventListener('resize', this.onWindowResized);
        }
        // https://github.com/ant-design/ant-design/issues/7191
        this.innerSlider = this.slick && this.slick.innerSlider;
        this.slickDOM = findDOMNode(this.slick);
        if (autoplay && dotsTimer) {
            var aniName_1 = dotsPosition == 'left' || dotsPosition == 'right' ? 'dotsAniVertical' : 'dotsAni', timerEl = this.slickDOM.querySelector('.timer');
            !!timerEl &&
                timerEl.style.setProperty('--dots-ani', aniName_1 + " " + autoplaySpeed / 1000 + "s infinite");
            this.slickDOM.addEventListener('mouseover', function () {
                var timerEl = _this.slickDOM.querySelector('.timer');
                !!timerEl && timerEl.style.setProperty('--dots-ani', 'none');
            });
            this.slickDOM.addEventListener('mouseout', function () {
                var timerEl = _this.slickDOM.querySelector('.timer');
                !!timerEl &&
                    timerEl.style.setProperty('--dots-ani', aniName_1 + " " + autoplaySpeed / 1000 + "s infinite");
            });
        }
    };
    Carousel.prototype.componentWillUnmount = function () {
        var autoplay = this.props.autoplay;
        if (autoplay) {
            window.removeEventListener('resize', this.onWindowResized);
            this.onWindowResized.cancel();
        }
    };
    Carousel.prototype.next = function () {
        this.slick.slickNext();
    };
    Carousel.prototype.prev = function () {
        this.slick.slickPrev();
    };
    Carousel.prototype.goTo = function (slide) {
        this.slick.slickGoTo(slide);
    };
    Carousel.prototype.render = function () {
        var _a = this.props, prefixCls = _a.prefixCls, className = _a.className, style = _a.style, dotsPosition = _a.dotsPosition, dotsTimer = _a.dotsTimer, nextArrow = _a.nextArrow, prevArrow = _a.prevArrow, centerMode = _a.centerMode, centerPadding = _a.centerPadding, slidesToShow = _a.slidesToShow, restProps = __rest(_a, ["prefixCls", "className", "style", "dotsPosition", "dotsTimer", "nextArrow", "prevArrow", "centerMode", "centerPadding", "slidesToShow"]);
        var cls = classNames(prefixCls, prefixCls + "-" + dotsPosition, className);
        var dotsCls = classNames('slick-dots', {
            'slick-dots-vertical': dotsPosition == 'left' || dotsPosition == 'right',
            timer: restProps.autoplay && dotsTimer
        });
        if (restProps.effect === 'fade') {
            restProps.fade = true;
        }
        return (React.createElement("div", { className: cls, style: style },
            React.createElement(SlickCarousel, __assign({}, restProps, { ref: this.saveSlick, nextArrow: nextArrow ? nextArrow : React.createElement(CustomArrow, { className: "slick-next", onClick: this.next }), prevArrow: prevArrow ? prevArrow : React.createElement(CustomArrow, { className: "slick-prev", onClick: this.prev }), centerMode: centerMode, centerPadding: centerPadding, slidesToShow: slidesToShow, dotsClass: dotsCls }))));
    };
    Carousel.defaultProps = {
        dots: true,
        arrows: false,
        prefixCls: 'fishd-carousel',
        draggable: false,
        dotsTimer: false,
        autoplaySpeed: 3000,
        dotsPosition: 'bottom',
        centerMode: false,
        centerPadding: '50px',
        slidesToShow: 1
    };
    return Carousel;
}(React.Component));
export default Carousel;
