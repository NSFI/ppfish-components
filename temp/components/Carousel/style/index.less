@import (reference) "../../../assets/css/index";

.@{fishd-prefix}-carousel {
  .reset-component;

  .slick-slider {
    position: relative;
    display: block;
    box-sizing: border-box;
    -webkit-touch-callout: none;
    -ms-touch-action: pan-y;
    touch-action: pan-y;
    -webkit-tap-highlight-color: transparent;
  }
  .slick-list {
    position: relative;
    overflow: hidden;
    display: block;
    margin: 0;
    padding: 0;

    &:focus {
      outline: none;
    }

    &.dragging {
      cursor: pointer;
    }
  }
  .slick-slider .slick-track,
  .slick-slider .slick-list {
    transform: translate3d(0, 0, 0);
  }

  .slick-track {
    position: relative;
    left: 0;
    top: 0;
    display: block;

    &:before,
    &:after {
      content: "";
      display: table;
    }

    &:after {
      clear: both;
    }

    .slick-loading & {
      visibility: hidden;
    }
  }
  .slick-slide {
    display: none;
    float: left;
    min-height: 1px;

    [dir="rtl"] & {
      float: right;
    }

    img {
      display: block;
    }

    &.slick-loading img {
      display: none;
    }

    &.dragging img {
      pointer-events: none;
    }
  }

  .slick-initialized .slick-slide {
    display: block;
  }

  .slick-loading .slick-slide {
    visibility: hidden;
  }

  .slick-vertical .slick-slide {
    display: block;
    height: auto;
    border: @border-width-base @border-style-base transparent;
  }
  .slick-arrow.slick-hidden {
    display: none;
  }

  // Arrows
  .slick-prev,
  .slick-next {
    position: absolute;
    display: flex;
    align-items: center;
    justify-content: center;
    height: 32px;
    width: 32px;
    cursor: pointer;
    background: rgba(0,0,0,0.2);
    color: #fff;
    top: 50%;
    transform: translateY(-50%);
    padding: 0;
    border: 0;
    outline: none;
    border-radius: 100%;
    z-index: 1;
    &:before {
      opacity: 0.6;
    }
    &:hover,
    &:focus {
      outline: none;
      &:before {
        opacity: 1;
      }
    }
    &.slick-disabled:before {
      opacity: 0.1;
    }
    .fishdicon;
  }

  .slick-prev {
    left: 20px;
    transform: translateY(-50%) rotate(180deg);
    &:before {
      content: "\e6e7";
    }
  }

  .slick-next {
    right: 20px;
    &:before {
      content: "\e6e7";
    }
  }
}

.slick-dots {
  position: absolute;
  list-style: none;
  display: block;
  text-align: center;
  margin: 0;
  padding: 0;
  width: 100%;
  height: @carousel-dot-height;
  li {
    position: relative;
    display: inline-block;
    vertical-align: top;
    text-align: center;
    margin: 0 2px;
    padding: 0;
  }

  li button {
    border: 0;
    cursor: pointer;
    background: #fff;
    opacity: 0.3;
    display: block;
    width: @carousel-dot-width;
    height: @carousel-dot-height;
    border-radius: 1px;
    outline: none;
    font-size: 0;
    color: transparent;
    transition: all 0.5s;
    padding: 0;
    &:hover,
    &:focus {
      opacity: 0.75;
    }
  }

  li.slick-active button {
    background: #fff;
    opacity: 1;
    width: @carousel-dot-active-width;
    &:hover,
    &:focus {
      opacity: 1;
    }
  }
}

.slick-dots.slick-dots-vertical {
  width: @carousel-dot-height;
  bottom: auto;
  top: 50%;
  transform: translateY(-50%);
  height: auto;

  li {
    display: block;
    margin: 4px 0;
  }

  li button {
    width: @carousel-dot-height;
    height: @carousel-dot-width;
  }

  li.slick-active button {
    width: @carousel-dot-height;
    height: @carousel-dot-active-width;
  }
}

.@{fishd-prefix}-carousel-top {
  .slick-dots {
    top: 12px;
  }
}

.@{fishd-prefix}-carousel-bottom {
  .slick-dots {
    bottom: 12px;
  }
}

.@{fishd-prefix}-carousel-right {
  .slick-dots-vertical {
    right: 12px;
  }
}

.@{fishd-prefix}-carousel-left {
  .slick-dots-vertical {
    left: 12px;
  }
}

.slick-dots.timer li {
  &.slick-active button {
    background: rgba(255,255,255,0.3);

    &:after {
      position: absolute;
      content: '';
      top: 0;
      left: 0;
      height: 100%;
      background: #fff;
      animation: var(--dots-ani, dotsAni 3s infinite);
    }
  }
}

.slick-dots-vertical.timer li {
  &.slick-active button {
    background: rgba(255,255,255,0.3);

    &:after {
      position: absolute;
      content: '';
      top: 0;
      left: 0;
      width: 100%;
      background: #fff;
      animation: var(--dots-ani, dotsAniVertical 3s infinite);
    }
  }
}

@keyframes dotsAni {
  from { width: 0; }
  to { width: 100%; }
}

@keyframes dotsAniVertical {
  from { height: 0; }
  to { height: 100%; }
}
