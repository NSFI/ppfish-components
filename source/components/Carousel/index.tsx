import Carousel from './Carousel';
import './style/index.less';

export { CarouselEffect, DotsPosition, CarouselProps, CarouselRef } from './Carousel';
export default Carousel;
