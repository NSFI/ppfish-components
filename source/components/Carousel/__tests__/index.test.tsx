import React from 'react';
import { mount } from 'enzyme';
import Carousel, { CarouselRef } from '../index';

const sleep = (timeout = 500) => new Promise(resolve => setTimeout(resolve, timeout));

describe('Carousel', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  it('should has innerSlider', () => {
    const ref = React.createRef<CarouselRef>();
    const wrapper = mount(
      <Carousel ref={ref}>
        <div />
      </Carousel>,
    );
    const { innerSlider } = ref.current;
    expect(typeof innerSlider.slickNext).toBe('function');
  });

  it('should has prev, next and go function', async () => {
    // jest.useRealTimers();
    const ref = React.createRef<CarouselRef>();
    const wrapper = mount(
      <Carousel speed={1} ref={ref} dots={false}>
        <div>1</div>
        <div>2</div>
        <div>3</div>
      </Carousel>,
    );
    const { prev, next, goTo } = ref.current;
    expect(typeof prev).toBe('function');
    expect(typeof next).toBe('function');
    expect(typeof goTo).toBe('function');
    expect(ref.current.innerSlider.state.currentSlide).toBe(0);
    goTo(2);
    // await sleep();
    jest.runAllTimers();
    expect(ref.current.innerSlider.state.currentSlide).toBe(2);
    ref.current.prev();
    // await sleep();
    jest.runAllTimers();
    expect(ref.current.innerSlider.state.currentSlide).toBe(1);
    next();
    jest.runAllTimers();
    // await sleep();
    expect(ref.current.innerSlider.state.currentSlide).toBe(2);
  });

  it('should trigger autoPlay after window resize', async () => {
    const ref = React.createRef<CarouselRef>();

    jest.useRealTimers();
    const wrapper = mount(
      <Carousel ref={ref} autoplay>
        <div>1</div>
        <div>2</div>
        <div>3</div>
      </Carousel>,
    );
    const spy = jest.spyOn(ref.current.innerSlider, 'autoPlay');
    window.resizeTo(1000, 0);
    expect(spy).not.toHaveBeenCalled();
    await new Promise(resolve => setTimeout(resolve, 500));
    expect(spy).toHaveBeenCalled();
  });

  it('cancel resize listener when unmount', async () => {
    const wrapper = mount(
      <Carousel autoplay>
        <div>1</div>
        <div>2</div>
        <div>3</div>
      </Carousel>,
    );
    const spy = jest.spyOn(window, 'removeEventListener');
    wrapper.unmount();
    expect(spy).toHaveBeenCalled();
  });

  describe('should works for dotPosition', () => {
    ['left', 'right', 'top', 'bottom'].forEach(
      (dotPosition: 'left' | 'right' | 'top' | 'bottom') => {
        it(dotPosition, () => {
          const wrapper = mount(
            <Carousel dotsPosition={dotPosition}>
              <div />
            </Carousel>,
          );
          expect(wrapper.render()).toMatchSnapshot();
        });
      },
    );
  });

  describe('should active when children change', () => {
    it('should active', () => {
      const wrapper = mount(<Carousel />);
      wrapper.setProps({
        children: <div />,
      });
      wrapper.update();
      expect(wrapper.find('.slick-active').length).toBeTruthy();
    });

    it('should keep initialSlide', () => {
      // react unsafe lifecycle don't works in React 15
      // https://github.com/akiran/react-slick/commit/97988e897750e1d8f7b10a86b655f50d75d38298
      if (process.env.REACT === '15') {
        return;
      }
      const wrapper = mount(<Carousel initialSlide={1} />);
      wrapper.setProps({
        children: [<div key="1" />, <div key="2" />, <div key="3" />],
      });
      wrapper.update();
      expect(wrapper.find('.slick-dots li').at(1).hasClass('slick-active')).toBeTruthy();
    });
  });
});
