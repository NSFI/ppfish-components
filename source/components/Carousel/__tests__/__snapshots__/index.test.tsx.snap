// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should works for dotPosition bottom 1`] = `
<div
  class="fishd-carousel fishd-carousel-bottom"
>
  <div
    class="slick-slider slick-initialized"
  >
    <div
      class="slick-list"
    >
      <div
        class="slick-track"
        style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
      >
        <div
          aria-hidden="false"
          class="slick-slide slick-active slick-current"
          data-index="0"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Carousel should works for dotPosition left 1`] = `
<div
  class="fishd-carousel fishd-carousel-left"
>
  <div
    class="slick-slider slick-initialized"
  >
    <div
      class="slick-list"
    >
      <div
        class="slick-track"
        style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
      >
        <div
          aria-hidden="false"
          class="slick-slide slick-active slick-current"
          data-index="0"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Carousel should works for dotPosition right 1`] = `
<div
  class="fishd-carousel fishd-carousel-right"
>
  <div
    class="slick-slider slick-initialized"
  >
    <div
      class="slick-list"
    >
      <div
        class="slick-track"
        style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
      >
        <div
          aria-hidden="false"
          class="slick-slide slick-active slick-current"
          data-index="0"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Carousel should works for dotPosition top 1`] = `
<div
  class="fishd-carousel fishd-carousel-top"
>
  <div
    class="slick-slider slick-initialized"
  >
    <div
      class="slick-list"
    >
      <div
        class="slick-track"
        style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
      >
        <div
          aria-hidden="false"
          class="slick-slide slick-active slick-current"
          data-index="0"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
